tests/pfc/test_unknown_mac.py

```diff
@@ -0,0 +1,344 @@
+              'pc_intfs': pc_intfs,
+              'ptf_pc_ports': ptf_pc_ports
+            }
+    yield setup
+
+@pytest.fixture
+def flushArpFdb(duthosts, rand_one_dut_hostname):
+    """
+    Fixture to flush all ARP and FDB entries
+
+    Args:
+        duthosts(AnsibleHost) : multi dut instance
+        rand_one_dut_hostname(AnsibleHost) : one of the dut instances from the multi dut
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    logger.info("Clear all ARP and FDB entries on the DUT")
+    duthost.shell("sonic-clear fdb all")
+    duthost.shell("ip neigh flush all")
+
+    yield
+
+    logger.info("Clear all ARP and FDB entries on the DUT")
+    duthost.shell("sonic-clear fdb all")
+    duthost.shell("ip neigh flush all")
+
+@pytest.fixture(autouse=True)
+def populateArp(unknownMacSetup, flushArpFdb, ptfhost, duthosts, rand_one_dut_hostname):
+    """
+    Fixture to populate ARP entry on the DUT for the traffic destination
+
+    Args:
+        unknownMacSetup(fixture) : module scope autouse fixture for test setup
+        flushArpFdb(fixture) : func scope fixture
+        ptfhost(AnsibleHost) : ptf host instance
+        duthosts(AnsibleHost) : multi dut instance
+        rand_one_dut_hostname(AnsibleHost) : one of the dut instances from the multi dut
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    setup = unknownMacSetup
+    ptfhost.script("./scripts/remove_ip.sh")
+    logger.info("Populate ARP entry for dest port")
+    ptfhost.command("ifconfig eth{} {}".format(setup['ptf_dst_port'], setup['vlan']['ips'][0]))
+    ptfhost.command("ping {} -c 3".format(setup['vlan']['addr']))
+
+    yield
+
+    logger.info("Clean up all ips on the PTF")
+    ptfhost.script("./scripts/remove_ip.sh")
+
+
+class PreTestVerify(object):
+    """ Verify ARP and FDB entries are populated correctly """
+    def __init__(self, duthost, dst_ip, dst_port):
+        """
+        Args:
+            duthost(AnsibleHost) : dut instance
+            dst_ip(string): traffic dest ip
+            dst_port(int): ptf id for the dest port
+        """
+        self.duthost = duthost
+        self.dst_ip = dst_ip
+        self.dst_port = dst_port
+        self.arp_entry = dict()
+
+    def _checkArpEntry(self):
+        """
+        Check if the ARP entry is present and populate the ARP to mac mapping
+        """
+        logger.info("Verify if the ARP entry is present for {}".format(self.dst_ip))
+        result = self.duthost.shell("ip neigh show {}".format(self.dst_ip))
+        pytest_assert(result['stdout_lines'], "{} not in arp table".format(self.dst_ip))
+        match = re.match("{}.*lladdr\s+(.*)\s+[A-Z]+".format(self.dst_ip),
+                         result['stdout_lines'][0])
+        pytest_assert(match,
+                      "Regex failed while retreiving arp entry for {}".format(self.dst_ip))
+        self.arp_entry.update({self.dst_ip : match.group(1)})
+
+    def _checkFdbEntry(self):
```

Comment:
Can we rename this function to _checkFdbEntryMiss

Reply:
done